packages/core/parcel-bundler/src/builtins/hmr-runtime.js

```diff
@@ -29,21 +29,36 @@ if ((!parent || !parent.isParcelRequire) && typeof WebSocket !== 'undefined') {
   var ws = new WebSocket(protocol + '://' + hostname + ':' + process.env.HMR_PORT + '/');
   ws.onmessage = function(event) {
     updatedAssets = {};
-    
+
     var data = JSON.parse(event.data);
 
     if (data.type === 'update') {
-      console.clear();
-
-      data.assets.forEach(function (asset) {
-        hmrApply(global.parcelRequire, asset);
+      var handled = false;
+      data.assets.forEach(function(asset) {
+        handled =
+          handled ||
+          (global.parcelRequire.cache[asset.id] &&
+            (Boolean(global.parcelRequire.cache[asset.id].hot._acceptCallbacks.length) ||
+             Boolean(global.parcelRequire.cache[asset.id].hot._disposeCallbacks.length)
+             )
```

Comment:
This isn't quite enough. You need to traverse upward from the asset to the root asset and check if any of the parents accept the update as well as the asset itself. Basically, we need to split `hmrAccept` up into two functions. One that traverses upward and checks if an update should be accepted, but without actually re-executing them. It can also keep track of all of the assets that need to be re-executed in a list. Then, if an update is accepted, go through the list we tracked earlier, and re-execute them.

Reply:
Done: extracted the actual execution into `hmrAcceptRun` and kept track of them in `assetsToAccept`